web/components/templates/alerts/alertForm.tsx

```diff
@@ -37,16 +44,32 @@ interface AlertFormProps {
 
 const AlertForm = (props: AlertFormProps) => {
   const { handleSubmit, onCancel, initialValues } = props;
+  const [slackRedirectUrl, setSlackRedirectUrl] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (window) {
+      setSlackRedirectUrl(`${window.location.origin}/slack/redirect`);
+    }
+  }, []);
```

Comment:
This pattern can be reduced to a `useMemo`
```suggestion
  const slackRedirectUrl = useMemo(() => {
    if (window) {
      return`${window.location.origin}/slack/redirect`;
    }
    return null;
  }, []);
```

HOWEVER, using window.location.orgin can be drilling in from the static props that get passed from the parent component:

[see this stack overflow](https://stackoverflow.com/questions/65199051/how-to-get-page-url-or-hostname-in-nextjs-project)

Reply:
@chitalian This is leading to passing the url through 4 components. Do we want to do that?

Reply:
@kavinvalli good call out. it's fine - let's maybe just useMemo instead?